scripts/openstack/progress.py

```diff
@@ -0,0 +1,47 @@
+"""Script for getting status of active openstack predicator experiments.
+
+WARNING: Will return the status of any other python3.8 processes running on
+        each machine in machines.txt
+
+See launch.py for information about the format of machines.txt.
+
+Usage example:
+    python scripts/openstack/progress.py --machines machines.txt \
+        --sshkey ~/.ssh/cloud.key
+"""
+
+import argparse
+import os
+
+from predicators.scripts.openstack.launch import run_cmds_on_machine
+
+
+def _main() -> None:
+    # Set up argparse.
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--machines", required=True, type=str)
+    parser.add_argument("--sshkey", required=True, type=str)
+    args = parser.parse_args()
+    openstack_dir = os.path.dirname(os.path.realpath(__file__))
+    # Load the machine IPs.
+    machine_file = os.path.join(openstack_dir, args.machines)
+    with open(machine_file, "r", encoding="utf-8") as f:
+        machines = f.read().splitlines()
+    # Make sure that the ssh key exists.
+    assert os.path.exists(args.sshkey)
+    # Loop through each machine and print those that have a python3.8 process.
+    progress_cmd = "pgrep python3.8"
```

Comment:
is there a reason you're only checking for python 3.8 in particular? what if someone is running 3.9 or 3.7? what would go wrong if you just do "pgrep python" here?

Reply:
launch.py specifically chooses python3.8 for the machines, so it shouldn't be an issue.